virtual-setup: add an initial set of Slack procedures (#178)
AlexsLemonade/training-modules: virtual-setup/slack-procedures.md

If you are not added to the training specific channel by 3 days prior to the start of training or are having trouble getting started with Slack, please [direct message](#using-direct-messages-during-training) a CCDL staff member (`Chante Bethell`, `Candace Savonen`, `Josh Shapiro` or `Jaclyn N. Taroni`) in Cancer Data Science Slack or email [training@ccdatalab.org](mailto:training@ccdatalab.org).

### Using the training-specific channel

You can use the training-specific private channel to do the following:

* Post questions during lectures
* Post errors and get help with debugging
``
* Treat the channel as a study group – if you get stuck on an exercise notebook or your own data, post something so other participants and instructors can help you.

If you have a question, it's very likely someone else in the course has the same question! 

**Responses to questions should be in [threads](https://slack.com/help/articles/115000769927-Use-threads-to-organize-discussions-) as much as possible**, including any screenshots or images. 
To start a thread, hover over the message you want to respond to and click the start a thread icon that appears on the top right of the message:

> <img src = screenshots/slack-start-a-thread.png width = 300>

Reviewer: I think we should show it in the full context. i.e including the message.
ex:
<img width="459" alt="Screen Shot 2020-04-15 at 9 50 48 AM" src="https://user-images.githubusercontent.com/15315514/79345070-a009a100-7efe-11ea-962d-dad3510e6f0c.png">

* The most difficult or confusing point of the module ("muddiest point"), which is posted anonymously to the channel
* What you liked about the module (anonymous, only instructors will see your answer)
* How we can improve the module (anonymous, only instructors will see your answer)

We will also ask you what you want to work on the next day on days _preceding_ consultation days.
Your answers will be sent to course instructors only and are **not** anonymous.
This allows instructors to group course participants based on their interests.

Use the `Submit Response` button in a Polly message to enter your feedback:

<img src = screenshots/slack-polly-response.png width = 500>

### Using Slack calling during training

If you need help during instruction or on consultation days and it is helpful to share your screen, a course instructor will initiate a [Slack call](https://slack.com/help/articles/216771908-Make-calls-in-Slack).
Your instructor will ask you to [share your screen](https://slack.com/help/articles/115003501303-Share-your-screen-with-Slack-Calls) using the share your screen icon circled in yellow below.

> 

Reviewer: The button is located at the bottom middle of your slack call window?

<img src = screenshots/slack-call-control.png width = 300>

_Note: Where possible please utilize headphones throughout the training workshop to minimize audio feedback._

### Using direct messages during training

> If you have a question that is _highly specific_ to your own data, you may [direct message](https://slack.com/help/articles/212281468-What-is-a-direct-message) a CCDL staff member. 

Reviewer: I would also bold `highly specific`

First, use the new message button in the top right side corner of the Slack interface.

<img src = screenshots/slack-compose-new-message.png width=300>

You are then able to search for the CCDL instructors – `Chante Bethell`, `Candace Savonen`, `Josh Shapiro` or `Jaclyn N. Taroni` – and compose your message.

Course instructors may direct you to the training-specific channel for more general questions or to another instructor where appropriate.

## General Slack use

In this section, we include a collection of tips that you may find helpful for using throughout training that were not covered above.

> 

Reviewer: I'm not sure if the formatting bar is visible by default. Do you think we should mention how to toggle it visible if they do not see it?

Author: How _does_ one toggle it if it's not visible? I don't know how to do that.

Reviewer: Oh you hit the `Aa`
![2020-04-15 10 55 42](https://user-images.githubusercontent.com/15315514/79352352-b9fbb180-7f07-11ea-8c58-c54448e65fd0.gif)

#### Attaching a file or image

You can [add a file or image to Slack](https://slack.com/help/articles/201330736-Add-files-to-Slack) by dragging a file to the Slack window or by using the paper clip icon next to the message field:

<img src = screenshots/slack-add-a-file.png width = 500> 

#### Adding code blocks to messages 

> Code or errors are usually easier to read when you use code formatting in Slack.

Reviewer 2: Changed my mind about wording here.
```suggestion
Code and error messages are usually easier to read when you use code formatting in Slack.
```

You are then able to search for the CCDL instructors – `Chante Bethell`, `Candace Savonen`, `Josh Shapiro` or `Jaclyn N. Taroni` – and compose your message.

Course instructors may direct you to the training-specific channel for more general questions or to another instructor where appropriate.

## General Slack use

In this section, we include a collection of tips that you may find helpful for using throughout training that were not covered above.

#### Attaching a file or image

You can [add a file or image to Slack](https://slack.com/help/articles/201330736-Add-files-to-Slack) by dragging a file to the Slack window or by using the paper clip icon next to the message field:

<img src = screenshots/slack-add-a-file.png width = 500> 

#### Adding code blocks to messages 

Code and error messages are usually easier to read when you use code formatting in Slack.
You can either select the code block icon below the message field:

<img src = screenshots/slack-code-block-icon.png width = 500>

Or you can type three backticks <code>```</code> into the message field and formatting will be applied:

<img src = screenshots/slack-backticks.gif width = 500 >

> You are then ready to paste your code or error message into the grey code block.

Reviewer: Should we add a line at the end about reaching out to us if they have questions or having trouble working with some of the features?

#### Instructions for setting up Slack

We have instructions for setting up Slack for your operation system available.

* [Linux](https://github.com/AlexsLemonade/training-modules/blob/master/virtual-setup/linux-instructions.md#slack)
* [Mac](https://github.com/AlexsLemonade/training-modules/blob/master/virtual-setup/mac-instructions.md#slack)
* [Windows](https://github.com/AlexsLemonade/training-modules/blob/master/virtual-setup/windows-instructions.md#slack)

#### Table of Contents  

- [How we use Slack during workshops](#how-we-use-slack-during-workshops)
  - [Using the training-specific channel](#using-the-training-specific-channel)
  - [Using Slack calling during training](#using-slack-calling-during-training)
  - [Using direct messages during training](#using-direct-messages-during-training)
- [General Slack use](#general-slack-use)
    - [Attaching a file or image](#attaching-a-file-or-image)
    - [Adding code blocks to messages](#adding-code-blocks-to-messages)

## How we use Slack during workshops

Course instructors will add you to a private channel specific to your training a few days before the start of your training workshop.

> If you are not added to the training specific channel or are having trouble getting started with Slack, please [direct message](#using-direct-messages-during-training) a CCDL staff member (`Chante Bethell`, `Candace Savonen`, `Josh Shapiro` or `Jaclyn N. Taroni`) in Cancer Data Science Slack or email [training@ccdatalab.org](mailto:training@ccdatalab.org).

Reviewer 2: Should we add a timeframe here? "If you are not added to the channel by X days before the start of training..."

Author: 3 days?

Reviewer 2: Sounds reasonable to me.

### Using the training-specific channel

You can use the training-specific private channel to do the following:

* Post questions during lectures

> * Post errors and get debug help

Reviewer 2: ```suggestion
* Post errors and get help with debugging
``

<img src = screenshots/slack-call-control.png width = 300>

_Note: Where possible please utilize headphones throughout the training workshop to minimize audio feedback._

### Using direct messages during training

If you have a question that is _highly specific_ to your own data, you may [direct message](https://slack.com/help/articles/212281468-What-is-a-direct-message) a CCDL staff member. 

First, use the new message button in the top right side corner of the Slack interface.

<img src = screenshots/slack-compose-new-message.png width=300>

You are then able to search for the CCDL instructors – `Chante Bethell`, `Candace Savonen`, `Josh Shapiro` or `Jaclyn N. Taroni` – and compose your message.

Course instructors may direct you to the training-specific channel for more general questions or to another instructor where appropriate.

## General Slack use

In this section, we include a collection of tips that you may find helpful for using throughout training that were not covered above.

#### Attaching a file or image

> You can [add a file or image to Slack](https://slack.com/help/articles/201330736-Add-files-to-Slack) using the paper clip icon next to the message field:

Reviewer 2: ```suggestion
You can [add a file or image to Slack](https://slack.com/help/articles/201330736-Add-files-to-Slack) by dragging a file to the Slack window or by using the paper clip icon next to the message field:
```

<img src = screenshots/slack-add-a-file.png width = 500> 

#### Adding code blocks to messages 

> Sometimes code or errors are easier to read when you use code formatting in Slack.

Reviewer 2: ```suggestion
Code or errors are usually easier to read when you use code formatting in Slack.
```
